lib/hanami/cli/registry.rb

```diff
@@ -86,6 +86,24 @@ def get(arguments)
         @commands.get(arguments)
       end
 
+      # @since x.x.x
+      # @api private
+      def before(command, &callback)
+        command_class(command).before(&callback)
```

Comment:
We can't delegate to a class the callbacks. If a command gets replaced, the attached callbacks won't be executed.

Let's say we have a `foo` gem with a CLI:

```ruby
module Foo
  module CLI
    module Commands
      extend Hanami::CLI::Registry

      class Console < Hanami::CLI::Command
        # ...
      end
    end
  end
end

Foo::CLI::Commands.register "console", Foo::CLI::Commands::Console
```

Then we have a `foo-greeting` gem, that provides a callback:

```ruby
Foo::CLI::Commands.after "console", { puts "goodbye" }
```

With this last operation, we have attached the callback to `Foo::CLI::Commands::Console`.

Now a third gem is loaded: `foo-pry`

```ruby
module Foo
  module Pry
    class Console < Hanami::CLI::Command
      # ...
    end
  end
end

Foo::CLI::Commands.register "console", Foo::Pry::Console
```

We have replaced the object that is invoked when we invoke `foo console` from shell.

But because callbacks were attached to the old class, they won't be executed.